dev/managedservicesplatform/spec/monitoring.go

```diff
@@ -99,3 +120,63 @@ func (r *ResponseCodeRatioAlertSpec) Validate() []error {
 
 	return errs
 }
+
+// CustomAlert
+type CustomAlert struct {
+	ID            string        `yaml:"-"`
+	Name          string        `yaml:"name"`
+	Description   string        `yaml:"description,omitempty"`
+	SeverityLevel SeverityLevel `yaml:"severityLevel"`
+	Duration      *string       `yaml:"duration,omitempty"`
+	MQLQuery      *string       `yaml:"mqlQuery,omitempty"`
+	PromQLQuery   *string       `yaml:"promQLQuery,omitempty"`
```

Comment:
I think we should have a more structured format:

```suggestion
    // Condition is required
	Condition struct {
	   // Type is one of 'mql' or 'promql'
	   Type string  `yaml:"type"`
	   Query string `yaml:"query"`
	   // TODO document what format duration is
	   DUration *string `yaml:"duration,omitempty"`
	}  `yaml:"condition"`
```
```

Reply:
This is more extensible, plus easier to understand when reading the spec definition and callsites (less implicit "it's either `MQLQuery` or `PromQLQuery`") and also makes validation better (`Condition` is required and cannot be nil or empty)